providers/storage/googleCloudDeltaStorage.js

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Google LLC. All rights reserved.
+// Licensed under the MIT License.
+
+const crypto = require('crypto');
+const moment = require('moment');
+const uuid = require('node-uuid');
+const Storage = require('@google-cloud/storage');
+
+class GoogleCloudDeltaStorage {
+  constructor(baseStore, bucketName, projectId, clientEmail, key, options) {
+    this.baseStore = baseStore;
+
+    // These values are used to keep delta files managed and unique
+    this.uniqueBlobId = uuid.v4(); // Avoid clashes in multiple processes environment
+    this.currentHour = moment.utc().format('HH');
+
+    this.bucketName = bucketName;
+    this.bucket = null;
+
+    this.options = options;
+
+    // Environment variables will cause new lines to be encoded as'\\n'
+    // which causes the google-cloud storage SDK to fail as it requires
+    // '\n' characters.
+    const parsedPrivateKey = key.replace(/\\n/g, '\n');
+
+    this.storage = new Storage({
+      projectId,
+      credentials: {
+        client_email: clientEmail,
+        private_key: parsedPrivateKey,
+      }
+    });
+
+  }
+
+  connect() {
+    return this.baseStore.connect()
+    .then(() => {
```

Comment:
formatting nit: General indenting pattern in the rest of the code is to have the `then` a level below the receiver. (so in a level on this line)

There are a few spots like this

Reply:
Updated in the delta and storage files.